ee/presencedetection/presencedetection_windows.go

```diff
@@ -0,0 +1,413 @@
+	if err != nil {
+		return false, fmt.Errorf("getting KeyCredentialManager from factory: %w", err)
+	}
+	defer managerObj.Release()
+	keyCredentialManager := (*KeyCredentialManager)(unsafe.Pointer(managerObj))
+
+	var isSupportedAsyncOperation *foundation.IAsyncOperation
+	ret, _, _ := syscall.SyscallN(
+		keyCredentialManager.VTable().IsSupportedAsync,
+		0, // Because this is a static function, we don't pass in a reference to `this`
+		uintptr(unsafe.Pointer(&isSupportedAsyncOperation)), // Windows.Foundation.IAsyncOperation<boolean>
+	)
+	if ret != 0 {
+		return false, fmt.Errorf("calling IsSupportedAsync: %w", ole.NewError(ret))
+	}
+
+	// IsSupportedAsync returns Windows.Foundation.IAsyncOperation<boolean>
+	iid := winrt.ParameterizedInstanceGUID(foundation.GUIDAsyncOperationCompletedHandler, booleanSignature)
+	statusChan := make(chan foundation.AsyncStatus)
+	handler := foundation.NewAsyncOperationCompletedHandler(ole.NewGUID(iid), func(instance *foundation.AsyncOperationCompletedHandler, asyncInfo *foundation.IAsyncOperation, asyncStatus foundation.AsyncStatus) {
+		statusChan <- asyncStatus
+	})
+	defer handler.Release()
+	isSupportedAsyncOperation.SetCompleted(handler)
+
+	select {
+	case operationStatus := <-statusChan:
+		if operationStatus != foundation.AsyncStatusCompleted {
+			return false, fmt.Errorf("IsSupportedAsync operation did not complete: status %d", operationStatus)
+		}
+	case <-time.After(5 * time.Second):
+		return false, errors.New("timed out waiting for IsSupportedAsync operation to complete")
+	}
+
+	res, err := isSupportedAsyncOperation.GetResults()
+	if err != nil {
+		return false, fmt.Errorf("getting results of IsSupportedAsync: %w", err)
+	}
+
+	return uintptr(res) > 0, nil
+}
+
+// register calls Windows.Security.Credentials.KeyCredentialManager.RequestCreateAsync.
+// It creates a new key credential for the current user and application.
+// See: https://learn.microsoft.com/en-us/uwp/api/windows.security.credentials.keycredentialmanager.requestcreateasync?view=winrt-26100
+func register(credentialName string) error {
+	// Get access to the KeyCredentialManager
+	factory, err := ole.RoGetActivationFactory("Windows.Security.Credentials.KeyCredentialManager", ole.IID_IInspectable)
+	if err != nil {
+		return fmt.Errorf("getting activation factory for KeyCredentialManager: %w", err)
+	}
+	defer factory.Release()
+	managerObj, err := factory.QueryInterface(keyCredentialManagerGuid)
+	if err != nil {
+		return fmt.Errorf("getting KeyCredentialManager from factory: %w", err)
+	}
+	defer managerObj.Release()
+	keyCredentialManager := (*KeyCredentialManager)(unsafe.Pointer(managerObj))
+
+	credentialNameHString, err := ole.NewHString(credentialName)
+	if err != nil {
+		return fmt.Errorf("creating credential name hstring: %w", err)
+	}
+	defer ole.DeleteHString(credentialNameHString)
+
+	var requestCreateAsyncOperation *foundation.IAsyncOperation
+	requestCreateReturn, _, _ := syscall.SyscallN(
```

Comment:
is the err worth keeping/noting here? or is that just not as informative as `ole.NewError(requestCreateReturn)`